src/main/kotlin/com/autonomousapps/tasks/ArtifactsReportTask.kt

```diff
@@ -29,6 +29,10 @@ abstract class ArtifactsReportTask : DefaultTask() {
     description = "Produces a report that lists all direct and transitive dependencies, along with their artifacts"
   }
 
+  /** To properly support the case when analyzed project module is a part of included build */
+  @get:Input
+  val rootDir: String = project.rootDir.absolutePath
```

Comment:
Thank you for the PR. I think that using the absolute file path for an input is probably overly aggressive. It will mean this task out would never come from the remote build cache on different machines.

Reply:
In the description I've mentioned, that this task has:
* `getClasspathArtifactFiles` Input with `PathSensitivity.ABSOLUTE` - isn't it the same? For each user it will be different as home dirs are not equal
* the generated `artifacts.json` has absolute File path:
```
...
  {
    "coordinates": {
      "type": "module",
      "identifier": "com.google.code.findbugs:jsr305",
      "resolvedVersion": "3.0.2",
      "gradleVariantIdentification": {
        "capabilities": [
          "com.google.code.findbugs:jsr305"
        ],
        "attributes": {}
      }
    },
    "file": "/Users/morph/.gradle/caches/modules-2/files-2.1/com.google.code.findbugs/jsr305/3.0.2/25ea2e8b0c338a877313bd4672d3fe056ea78f0d/jsr305-3.0.2.jar"
  },
...
```

So it seems that this new parameter will not make it worse. WDYT?

Reply:
I've updated the description and suggested one more option with relative path from current project to root.

Using absolute root path as input is motivated for reasons explained above, but still there is an opportunity to declare relative path from current project of task to root project (note: not vice versa). It means, that in most cases such Input parameter will have value like `".."` or `"../.."`. But for included builds it will look like `"../../included/composite-project"`.

Reply:
You're not wrong that it won't be worse, but nevertheless I would like to improve the situation. I have it on my agenda to inspect usages of this task more closely and see if we need to output the absolute path to the file. I'd like to improve the task's cacheability.

Reply:
This is good, I also think this task has potential for caching optimization.
I'd like to proceed with this PR. WDYT about changing absolute path to relative path here from current project to root project?
The comment above explains possible values. I'd like to add one more note here: in some special cases it's not possible to calculate relative path, e.g. for Windows if paths are on different logical disks. But in this case the absolute path can be used as a failover.
Should I proceed with this approach?